src/webapp/components/survey/hook/useGridSection.ts

```diff
@@ -28,7 +29,13 @@ export const useGridSection = (
         const speciesQuestion = speciesSection.questions.find(isSpeciesQuestion);
         if (speciesQuestion) {
             const options = getAntibioticOptions(speciesQuestion);
-            setGridOptions(options);
+            const optionsWithoutBlacklistedAntibiotics = options?.filter(
```

Comment:
this is business logic and does not belong in presentation layer.. The question visibility should be set to false in domain layer

Reply:
Please move this logic to custom hook useASTGuidelinesOptions() : getAntibioticOptions(). You can get the antibioticsList inside the custom hook by 
```
 const antibioticsBlacklist = currentModule?.rulesBySurvey?.find(
                rule => rule.surveyId === prevalenceCurrentFormId
            )?.antibioticBlacklist;
```
So there is no need to pass antibioticsBlackList as a prop to useGridSection, GridSection and remove changes to  SurveyForm